subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationNode.java

```diff
@@ -30,34 +28,41 @@
 import org.gradle.internal.operations.BuildOperationDescriptor;
 import org.gradle.internal.operations.BuildOperationExecutor;
 import org.gradle.internal.operations.RunnableBuildOperation;
+import org.gradle.util.Path;
 
 import javax.annotation.Nullable;
 import java.io.File;
 import java.util.Collections;
 import java.util.Set;
-import java.util.concurrent.atomic.AtomicInteger;
 
 public abstract class TransformationNode extends Node {
-    private static final AtomicInteger ORDER_COUNTER = new AtomicInteger();
 
-    private final int order = ORDER_COUNTER.incrementAndGet();
     protected final TransformationStep transformationStep;
     protected TransformationSubject transformedSubject;
+    private final TransformationIdentity identity;
 
     public static TransformationNode chained(TransformationStep current, TransformationNode previous, ExecutionGraphDependenciesResolver executionGraphDependenciesResolver) {
         return new ChainedTransformationNode(current, previous, executionGraphDependenciesResolver);
     }
 
-    public static TransformationNode initial(TransformationStep initial, ResolvableArtifact artifact, ExecutionGraphDependenciesResolver executionGraphDependenciesResolver) {
-        return new InitialTransformationNode(initial, artifact, executionGraphDependenciesResolver);
+    public static TransformationNode initial(TransformationStep initial, ResolvableArtifact artifact, ExecutionGraphDependenciesResolver executionGraphDependenciesResolver, Path buildPath) {
+        return new InitialTransformationNode(initial, artifact, executionGraphDependenciesResolver, buildPath);
     }
 
     protected TransformationNode(TransformationStep transformationStep) {
+        this.identity = TransformationIdentity.create();
         this.transformationStep = transformationStep;
     }
 
+    @Override
+    public TransformationIdentity getIdentity() {
+        return identity;
+    }
+
     public abstract void execute(BuildOperationExecutor buildOperationExecutor, ArtifactTransformListener transformListener);
 
+    protected abstract Path getBuildPath();
```

Comment:
Why not make this part of the `TransformationIdentity` instead?

Reply:
We already have this for `TaskIdentity`, so I don't see why this would cause more problems. Anyway, let's merge this, and we'll fix it later.